src/magnet_coils.py

```diff
@@ -0,0 +1,534 @@
+            volume_ids (range): new volume ids corresponding to magnet volumes
+                following cutting operation
+        """
+
+        # Define sweeping surface width
+        # Multiply by factor of 2 to be conservative
+        rec_width = 2*self.average_radial_distance
+
+        cubit.cmd(f'create surface rectangle width {rec_width} yplane')
+        surf_id = cubit.get_last_id("surface")
+
+        # Shift surface to positive x axis
+        cubit.cmd(f'move Surface {surf_id} x {rec_width/2}')
+
+        # Revolve surface to create wedge spanning toroidal extent
+        cubit.cmd(
+            (f'sweep surface {surf_id} zaxis angle '
+             f'{np.rad2deg(self.toroidal_extent)}')
+        )
+        sweep_id = cubit.get_last_id("volume")
+
+        # Remove magnets and magnet portions not within toroidal extent
+        cubit.cmd(
+            'intersect volume ' + ' '.join(str(i) for i in volume_ids)
+            + f' {sweep_id}'
+        )
+
+        # renumber volume ids from 1 to N
+        cubit.cmd('compress all')
+
+        # extract new volume ids
+        volume_ids = cubit.get_entities('volume')
+
+        return volume_ids
+
+    def build_magnet_coils(self):
+        """
+        Builds each filament in self.filtered_filaments in cubit, then
+        cuts to the toroidal extent using self._cut_magnets()
+        """
+
+        self.create_magnet_coils()
+
+        volume_ids = []
+
+        for magnet_coil in self.magnet_coils:
+            volume_id = magnet_coil.create_magnet()
+            volume_ids.append(volume_id)
+
+        volume_ids = self._cut_magnets(volume_ids)
+
+        self.volume_ids = volume_ids
+
+    def mesh_magnets(self):
+        """Creates tetrahedral mesh of magnet volumes, exports to
+        exodus and converts to h5m
+        """
+        # Mesh magnet volumes
+        for vol in self.volume_ids:
+            cubit.cmd(f'volume {vol} scheme tetmesh')
+            cubit.cmd(f'mesh volume {vol}')
+
+        self.logger.info('Exporting coil mesh...')
+
+        # Define export paths
+        exo_path = Path(self.export_dir) / 'coil_mesh.exo'
+        h5m_path = Path(self.export_dir) / 'coil_mesh.h5m'
+
+        # EXODUS export
+        cubit.cmd(f'export mesh "{exo_path}" overwrite')
+
+        # Convert EXODUS to H5M
+        subprocess.run(f'mbconvert {exo_path} {h5m_path}', shell=True)
+
+
+class MagnetCoil(object):
+    def __init__(
+        self,
+        filament_,
+        shape_,
+        shape_str_
+    ):
+        self.filament = filament_
+        self.shape = shape_
+        self.shape_str = shape_str_
+
+    def unit_vector(self, vec):
+        """Normalizes given vector.
+
+        Arguments:
+            vec (array of [float, float, float]): input vector to be normalized.
+
+        Returns:
+            vec (array of [float, float, float]): normalized input vector.
+        """
+        vec = normalize(vec.reshape(-1, 1), axis=0).ravel()
+
+        return vec
```

Comment:
We should figure out how to make this available to all of parastell instead of in this class.

Reply:
ill pop this out to be its own function that can get relocated and imported as desired once we figure out the structure